### PR TITLE
Fix limits event handling

### LINKED ISSUE #1339: Limits events on unmapped target causes error in Python DecomMicroservice
Disable and enable a limits in INST and check for error in INST2 decom microservice

### PR DESCRIPTION
**Pull request by jmthomas**
closes #1339

### CODE REVIEW

File: openc3/lib/openc3/topics/limits_event_topic.rb [hunk 1]
```diff
@@ -179,7 +179,7 @@ def self.sync_system(scope:)
             enabled = limits_settings['enabled']
             persistence = limits_settings['persistence_setting']
             limits_settings.each do |limits_set, settings|
-              next unless Hash === limits_set
+              next unless Hash === settings
```
Comment (jmthomas) [inline openc3/lib/openc3/topics/limits_event_topic.rb:182]: This was wrong but it was in the initial sync ... not sure the effects

File: openc3/python/test/topics/test_limits_event_topic.py [hunk 2]
```diff
@@ -81,3 +236,114 @@ def test_removes_all_items_from_the_out_of_limits_list(self):
...
+                "enabled": True,
+                "message": "TEST1",
+            }
+            LimitsEventTopic.write(event, scope="DEFAULT")
+            time.sleep(0.01)
+            getattr(LimitsEventTopic, method)(scope="DEFAULT")
+            self.assertEqual(System.limits.enabled("INST", "HEALTH_STATUS", "TEMP1"), True)
+
+            # TODO: System not getting set?
+            limits = System.limits.get("INST", "HEALTH_STATUS", "TEMP1")
```
Comment (jmthomas) [inline openc3/python/test/topics/test_limits_event_topic.py:315]: This doesn't work because I think the System isn't shared like in Ruby

**Review by jmthomas — COMMENTED** [review]
Python code still needs some work

File: openc3/python/openc3/packets/limits.py [hunk 3]
```diff
@@ -83,7 +83,7 @@ def get(self, target_name, packet_name, item_name, limits_set=None):
             if limits_set:
                 limits_set = str(limits_set).upper()
             else:
-                limits_set = self.system.limits_set
+                limits_set = self.system.limits_set()
```
Comment (jmthomas) [inline openc3/python/openc3/packets/limits.py:86]: I don't think `get` is called anywhere so this bug wasn't found

File: openc3/python/openc3/packets/limits.py [hunk 4]
```diff
@@ -159,6 +159,9 @@ def set(
         if green_low and green_high:
             limits_for_set[4] = float(green_low)
             limits_for_set[5] = float(green_high)
+        else:
+            limits_for_set[4] = None
+            limits_for_set[5] = None
```
Comment (jmthomas) [inline openc3/python/openc3/packets/limits.py:164]: Ruby calls `delete_at` if the existing limits has green low and green ... I think it's cleaner to just clear it in the else

**Review by jmthomas — COMMENTED** [review]
Lots of python bugs in the sync due to KeyError

File: openc3/python/openc3/topics/limits_event_topic.py [hunk 5]
```diff
@@ -239,12 +239,12 @@ def sync_system_thread_body(cls, scope, block_ms=None):
...
                     item_name = event["item_name"]
-                    target = telemetry[target_name]
+                    target = telemetry.get(target_name)
                     if target:
-                        packet = target[packet_name]
+                        packet = target.get(packet_name)
                         if packet:
-                            enabled = ConfigParser.handle_true_false_none(event["enabled"])
-                            persistence = event["persistence"]
+                            enabled = event.get("enabled", None)
```
Comment (ryanmelt) [inline openc3/python/openc3/topics/limits_event_topic.py:246]: Is the handle_true_false needed? ie.  does enabled every get set to false?

**Review by ryanmelt — APPROVED** [review]

File: openc3/python/openc3/topics/limits_event_topic.py (hunk 5 above)
Comment (jmthomas) [reply openc3/python/openc3/topics/limits_event_topic.py:246]: It gets set to `False` which is valid JSON that we decode. So we're always returning a boolean.

**Pull request merged**